### DISCUSSION TITLE
Searching in G-Node

### DISCUSSION THREAD

**Discussion by Codehunt14** [eclipse-glsp/glsp#1200] [Q&A]
Hi,
I'm working on implementing a Search functionality within the GLSP editor (GModel). I aim to create a feature similar to the search function (Ctrl+F) found in VS Code, but tailored to search within the GNodes.
I am also facing the issue in activating the search dialog box in GLSP editor.
What would be the approach to achieve this functionality?

**Comment by planger** [comment]
Hi @Codehunt14, thank you for your message! This sounds like a great feature.

Please note though, we already have a search feature which may at least partially already address your needs and maybe can be extended with a more sophisticated search behavior to match your requirements.
https://github.com/eclipse-glsp/glsp-client/tree/master/packages/client/src/features/accessibility/search

[a11y_2.webm](https://github.com/eclipse-glsp/glsp/assets/588090/2b32cf2e-7783-4735-9eb4-e98e1c88bbdb)

It is an optional module (disabled by default) and not particularly integrated into VS Code or Theia yet. However, we'd really like to do that. So if you are up to it, we'd appreciate your help on that.
Basically, what's missing is to hook up the keyboard bindings and correct activation (based on context keys) for the shortcuts to enable e.g. search and all the other keyboard features (moving elements, zooming, panning, navigation) with the native API of the respective tool platform, such as VS Code and Theia. This is to ensure the keybindings can be configured by the user and work reliably depending on the activation/focus state.

**Reply by planger** [reply]
For reference, this is the Github issue: https://github.com/eclipse-glsp/glsp/issues/1129
